polyply/tests/test_build_file_parser.py

```diff
@@ -181,11 +294,56 @@ def test_system():
 def test_parser(test_system, lines, tagged_mols, tagged_nodes):
    lines = textwrap.dedent(lines).splitlines()
    ff = vermouth.forcefield.ForceField(name='test_ff')
+   top = Topology(ff)
    polyply.src.build_file_parser.read_build_file(lines,
-                                                 test_system.molecules)
+                                                 test_system.molecules,
+                                                 top)
    for idx, mol in enumerate(test_system.molecules):
        for node in mol.nodes:
            if "restraints" in mol.nodes[node]:
                assert node in tagged_nodes
                assert idx in tagged_mols
 
+@staticmethod
+@pytest.mark.parametrize('lines, expected', (
+   # basic test
+   ("""
+   [ molecule ]
+   ; name from to
+   AA    0  2
+   ;
+   [ presistence_length ]
+   ; model  lp start stop
+     WCM   4.0  0    8
+   """,
+   [["WCM", 4.0, 0, 8, np.array([0, 1])]]),
+   # test two batches are recognized
+   ("""
+   [ molecule ]
+   ; name from to
+   AA    0  2
+   ;
+   [ presistence_length ]
+   ; model  lp start stop
+   WCM   4.0  0    8
+   [ molecule ]
+   ; name from to
+   BB 2  4
+   [ presistence_length ]
+   WCM   2.0   0  8
+   """,
+   [["WCM", 4.0, 0, 8, np.array([0, 1])],
+    ["WCM", 2.0, 0, 8, np.array([2, 3])]]
+   )))
```

Comment:
I'm not sure I read this correctly, but does `from=i, to=j` translate to `(i, j+1)`? How what why where? The combination of starting to count a 0, but including the final number looks super strange to me.

Reply:
Agreed but that's more a documentation/consistency problem. The molecule directive goes `from to` because a range is constructed, whereas the `persistence_length` directive refers to two nodes. I agree a bit bizzare